### ISSUE TITLE
putting the same artifact to two resources: file missing in second put?

### ISSUE THREAD

**Issue by evanshort73** [concourse/concourse#473]
We had a slack question: https://concourseci.slack.com/archives/general/p1465594650001612

"""

We're seeing some weird behavior with a task output being used by multiple put steps

So we have a task like this:
```
outputs:
- name: my-dir
```

It prints a value to a file in `my-dir/file`.

Then, we have two `put`'s in our pipeline, each of which uses the output of that task
 ```
- put: resource-1
  params:
    file_path: my-dir/file
- put: resource-2
  params:
    file_path: my-dir/file
```

The file is present in the first `put`, but not the second one

"""

**Comment by dsabeti** [comment]
Hey there! We're the ones who saw this problem. The `put` in question was a `put` to the bosh deployment resource, if that helps for trying to reproduce the error.

**Comment by vito** [comment]
Was the output name the same as the `put` name?

**Comment by ljfranklin** [comment]
We're running a similar issue after upgrading to Concourse v1.3.1. We have a task that produces an output called `director-config` with several files in it. The next task declares `director-config` as an input and is able to see all the expected files. However, a third task which also declares `director-config` as an input reports "No such file or directory" for the files in `director-config`. This seems to happen ~50% of our builds.

> Was the output name the same as the put name?

Nope, none of the task or resource names are called `director-config`, only the single output and two inputs listed above.

Any ideas for a workaround? We can try daisy-chaining the outputs (a -> b -> c) rather than passing a single output to two tasks (a -> b, a -> c) to see if that helps.

**Comment by freelock** [comment]
Hi,

I hit this same issue with my Matrix-Notification resource. In my usage I had 3 "put" steps defined for the same matrix-notification resource - one in on-failure, one in an ensure step, and one in a regular step.

The job I was trying to diagnose was failing, and so the on-failure notification got triggered, but then the ensure one failed to run, because the output directory passed to the container is empty for the 2nd run.

I worked around this (for now) by combining the messages into a json blob, and having my bot that listens to the matrix notification extract the data and route appropriately, and then I could eliminate the need for multiple resources -- now it's all handled by the single "ensure" step.

**Comment by cunnie** [comment]
@ljfranklin : it appears that the Concourse problem we were seeing was caused by a bad commit (i.e. it's not a Concourse problem after all — it's a bug we introduced in our code). @forrestsill & I will fix it Monday.

**Comment by vito** [comment]
I think this should be fixed as part of 1.4.x.

**Comment by cunnie** [comment]
Yeah, it appears fixed — we're not getting the errors anymore now that we've upgraded to 1.4.1.
